src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -53,8 +54,23 @@ void ov::hetero::CompiledModel::compile_model(const std::shared_ptr<ov::Model>&
         auto it_info = node_info.find("affinity");
         if (it_info != node_info.end()) {
             OPENVINO_ASSERT(it_info->second.is<std::string>(), "Unexpected type of \"affinity\" attribute");
-            query_model_result.emplace(node->get_friendly_name(), it_info->second.as<std::string>());
+            auto node_affinity = it_info->second.as<std::string>();
+            query_model_result.emplace(node->get_friendly_name(), node_affinity);
             user_set_affinities = true;
+            if (ov::op::util::is_parameter(node)) {
+                for (const auto& output : node->outputs()) {
+                    for (auto out_inputs : output.get_target_inputs()) {
+                        auto node_info = out_inputs.get_node()->get_rt_info();
+                        auto it_info = node_info.find("affinity");
+                        if (it_info->second.as<std::string>() != node_affinity) {
+                            auto convert = std::make_shared<ov::op::v0::Convert>(output, out_inputs.get_element_type());
+                            ov::copy_runtime_info(node, convert);
+                            out_inputs.replace_source_output(convert);
+                            query_model_result.emplace(convert->get_friendly_name(), node_affinity);
```

Comment:
for this specific case, maybe we can check the subgraphs, if we have parameter - > result subgraph and the result is an intermediate result , we can change the result output ptr instead of using the default input one?
by this, we can 1) avoid extra convert nodes 2) do not need to add convert to supported query result empirically, because convert may be not a supported node